src/main/scala/io/iohk/ethereum/utils/FutureOps.scala

```diff
@@ -0,0 +1,17 @@
+package io.iohk.ethereum.utils
+import akka.actor.Scheduler
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.duration.FiniteDuration
+
+object FutureOps {
+  implicit class FutureOps[T](future: Future[T]) {
+
+    def delayedBy(delay: FiniteDuration)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] =
+      future.flatMap(value => {
```

Comment:
maybe this doubled brackets are not needed

Reply:
AFAIK they are

Reply:
Or not really :) Fixed